### PR TITLE
first

### PR DESCRIPTION
**Pull request by voloshynamaryna11**

### CODE REVIEW

**Review by Victoria-Moiseienko — COMMENTED** [review]
Good job

File: src/main/java/internet/market/controllers/DeleteProductAdminController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import internet.market.lib.Injector;
+import internet.market.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteProductAdminController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/internet/market/controllers/DeleteProductAdminController.java]:
```suggestion
public class DeleteProductСontroller extends HttpServlet {
```

File: src/main/java/internet/market/controllers/GetAllOrdersController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+public class GetAllOrdersController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("internet.market");
+    private OrderService orderService = (OrderService) injector
+            .getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Order> all = orderService.getAll();
```
Comment (KseniiaMakarova) [inline src/main/java/internet/market/controllers/GetAllOrdersController.java]:
```suggestion
        List<Order> allOrders = orderService.getAll();
```

File: src/main/java/internet/market/controllers/GetAllProductsAdminController.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import internet.market.model.Product;
+import internet.market.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllProductsAdminController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/internet/market/controllers/GetAllProductsAdminController.java]:
```suggestion
public class ManageProductsController extends HttpServlet {
```

File: src/main/java/internet/market/controllers/GetAllProductsAdminController.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    private static final Injector injector = Injector.getInstance("internet.market");
+    private ProductService productService = (ProductService) injector
+            .getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Product> all = productService.getAll();
```
Comment (KseniiaMakarova) [inline src/main/java/internet/market/controllers/GetAllProductsAdminController.java]:
```suggestion
        List<Product> allProducts = productService.getAll();
```

File: src/main/java/internet/market/controllers/PlaceAnOrderController.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import internet.market.lib.Injector;
+import internet.market.service.OrderService;
+import internet.market.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class PlaceAnOrderController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/internet/market/controllers/PlaceAnOrderController.java]:
```suggestion
public class CompleteOrderController extends HttpServlet {
```

File: src/main/java/internet/market/model/Order.java [hunk 6]
```diff
@@ -7,12 +7,21 @@
     private Long id;
     private List<Product> products;
     private Long userId;
+    private double sum;
```
Comment (KseniiaMakarova) [inline src/main/java/internet/market/model/Order.java]: let's remove this field - we will have troubles later when adding a DB connection
if you wish you can sum all prices inside your JSP

File: src/main/java/internet/market/service/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -20,6 +21,8 @@
     @Override
     public Order completeOrder(ShoppingCart shoppingCart) {
         Order order = new Order(shoppingCart.getUserId());
+        order.setSum(shoppingCart.getProducts().stream()
+                .mapToDouble(Product::getPrice).sum());
```
Comment (KseniiaMakarova) [inline src/main/java/internet/market/service/impl/OrderServiceImpl.java]: let's remove this as well

File: src/main/webapp/WEB-INF/views/admin/allOrders.jsp [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+<head>
+    <title>All orders of User</title>
+</head>
+<body>
+<h1>All your orders</h1>
+<table border="1">
+    <tr>
+        <th>ID</th>
+        <th>UserId</th>
+        <th>Sum</th>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/admin/allOrders.jsp]: let's not display the sum here

File: src/main/webapp/WEB-INF/views/admin/allProducts.jsp [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            <td>
+                <c:out value="${product.price}"/>
+            </td>
+            <td>
+                <a href="${pageContext.request.contextPath}/admin/product/delete?id=${product.id}">Delete</a>
+            </td>
+        </tr>
+    </c:forEach>
+</table>
+<a href="${pageContext.request.contextPath}">Go to the main page</a>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/admin/allProducts.jsp:32]: let's also have a link to the form that adds a new product to the storage

File: src/main/webapp/WEB-INF/views/order/orderDetails.jsp [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+            <td>
+                <c:out value="${product.name}"/>
+            </td>
+            <td>
+                <c:out value="${product.price}"/>
+            </td>
+        </tr>
+    </c:forEach>
+</table>
+<a href="${pageContext.request.contextPath}">Back to the main page</a>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/orderDetails.jsp]: here you can calculate the sum if you wish

File: src/main/webapp/WEB-INF/views/order/ordersOfUser.jsp [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+<html>
+<head>
+    <title>All orders of User</title>
+</head>
+<body>
+<h1>All your orders</h1>
+<table border="1">
+    <tr>
+        <th>ID</th>
+        <th>Sum</th>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/ordersOfUser.jsp]:
```suggestion
        <th>ID</th>
```

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -92,4 +92,84 @@
...
     </servlet-mapping>
+
+    <servlet>
+        <servlet-name>placeAnOrder</servlet-name>
+        <servlet-class>internet.market.controllers.PlaceAnOrderController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>placeAnOrder</servlet-name>
+        <url-pattern>/cart/placeAnOrder</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/complete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>showUserOrders</servlet-name>
+        <servlet-class>internet.market.controllers.GetAllOrdersOfUserController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>showUserOrders</servlet-name>
+        <url-pattern>/order/user/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/user/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 14]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>showAllOrders</servlet-name>
+        <servlet-class>internet.market.controllers.GetAllOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>showAllOrders</servlet-name>
+        <url-pattern>/admin/order/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/admin/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteOrder</servlet-name>
+        <servlet-class>internet.market.controllers.DeleteOrderController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteOrder</servlet-name>
+        <url-pattern>/admin/order/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/admin/orders/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 16]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteProduct</servlet-name>
+        <servlet-class>internet.market.controllers.DeleteProductAdminController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteProduct</servlet-name>
+        <url-pattern>/admin/product/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/admin/products/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 17]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allProductsAdmin</servlet-name>
+        <servlet-class>internet.market.controllers.GetAllProductsAdminController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allProductsAdmin</servlet-name>
+        <url-pattern>/admin/product/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/admin/products</url-pattern>
```

File: src/main/webapp/web.xml [hunk 18]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteProductFromCart</servlet-name>
+        <servlet-class>internet.market.controllers.DeleteProductFromCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteProductFromCart</servlet-name>
+        <url-pattern>/cart/product/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cart/products/delete</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/internet/market/controllers/DeleteProductAdminController.java (hunk 1 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/DeleteProductAdminController.java]: done

File: src/main/java/internet/market/controllers/GetAllOrdersController.java (hunk 2 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/GetAllOrdersController.java]: done

File: src/main/java/internet/market/controllers/GetAllProductsAdminController.java (hunk 3 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/GetAllProductsAdminController.java]: done

File: src/main/java/internet/market/controllers/GetAllProductsAdminController.java (hunk 4 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/GetAllProductsAdminController.java]: done

File: src/main/java/internet/market/controllers/PlaceAnOrderController.java (hunk 5 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/PlaceAnOrderController.java]: done

File: src/main/java/internet/market/service/impl/OrderServiceImpl.java (hunk 7 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/service/impl/OrderServiceImpl.java]: done

File: src/main/webapp/WEB-INF/views/admin/allOrders.jsp (hunk 8 above)
Comment (voloshynamaryna11) [reply src/main/webapp/WEB-INF/views/admin/allOrders.jsp]: done

File: src/main/webapp/WEB-INF/views/admin/allProducts.jsp (hunk 9 above)
Comment (voloshynamaryna11) [reply src/main/webapp/WEB-INF/views/admin/allProducts.jsp:32]: done

File: src/main/webapp/WEB-INF/views/order/ordersOfUser.jsp (hunk 11 above)
Comment (voloshynamaryna11) [reply src/main/webapp/WEB-INF/views/order/ordersOfUser.jsp]: done

File: src/main/webapp/WEB-INF/views/order/orderDetails.jsp (hunk 10 above)
Comment (voloshynamaryna11) [reply src/main/webapp/WEB-INF/views/order/orderDetails.jsp]: done

File: src/main/java/internet/market/model/Order.java (hunk 6 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/model/Order.java]: done

File: src/main/webapp/web.xml (hunk 12 above)
Comment (voloshynamaryna11) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 18 above)
Comment (voloshynamaryna11) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 13 above)
Comment (voloshynamaryna11) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 14 above)
Comment (voloshynamaryna11) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 15 above)
Comment (voloshynamaryna11) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 16 above)
Comment (voloshynamaryna11) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 17 above)
Comment (voloshynamaryna11) [reply src/main/webapp/web.xml]: done

File: src/main/java/internet/market/controllers/GetOrderDetailsController.java [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetOrderDetailsController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("internet.market");
+    private OrderService orderService = (OrderService) injector
+            .getInstance(OrderService.class);
+    private ProductService productService = (ProductService) injector
+            .getInstance(ProductService.class);
```
Comment (KseniiaMakarova) [inline src/main/java/internet/market/controllers/GetOrderDetailsController.java]: do you use this?

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/internet/market/controllers/GetOrderDetailsController.java (hunk 19 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/GetOrderDetailsController.java]: done

**Pull request merged**